cpp/dolfinx/fem/CoordinateElement.h

```diff
@@ -148,17 +156,16 @@ class CoordinateElement
   /// @param[in,out] x The physical coordinates of the reference points X
   /// @param[in] cell_geometry The cell node coordinates (physical)
   /// @param[in] phi Tabulated basis functions at reference points X
-  static void push_forward(xt::xtensor<double, 2>& x,
-                           const xt::xtensor<double, 2>& cell_geometry,
-                           const xt::xtensor<double, 2>& phi);
+  template <typename U, typename V, typename W>
+  static void push_forward(U&& x, const V& cell_geometry, const W& phi)
+  {
+    for (std::size_t i = 0; i < x.extent(0); ++i)
+      for (std::size_t j = 0; j < x.extent(1); ++j)
+        x(i, j) = 0;
```

Comment:
Do we need to do the double loop here? 
Doesn't mdspan work with 
```c++
std::fill(x.begin(), x.end(), 0);
```
?

There was a similar instance of this in `math.h`.

Reply:
Nevermind, as we can't guarantee that the `mdspan` is contiguous in memory that doesn't work.

Reply:
No - `mdspan` doesn't have `begin/end`. We could add an optimisation later that checks if `x` has contiguous storage, in which case we could use `std::fill`.